apps/meteor/ee/server/lib/ldap/Manager.ts

```diff
@@ -553,18 +555,37 @@ export class LDAPEEManager extends LDAPManager {
 		});
 	}
 
-	private static async updateExistingUsers(ldap: LDAPConnection, converter: LDAPDataConverter): Promise<void> {
+	private static async reviewExistingUsers(
+		ldap: LDAPConnection,
+		converter: LDAPDataConverter,
+		settings: {
+			updateExistingUsers: boolean;
+			removeDeletedUsers: boolean;
+		},
+	): Promise<void> {
 		const users = await UsersRaw.findLDAPUsers().toArray();
+		const { updateExistingUsers, removeDeletedUsers } = settings;
+
 		for await (const user of users) {
 			const ldapUser = await this.findLDAPUser(ldap, user);
 
-			if (ldapUser) {
-				const userData = this.mapUserData(ldapUser, user.username);
-				converter.addUserSync(userData);
+			if (updateExistingUsers && ldapUser) {
+				this.updateExistingUser(ldapUser, user, converter);
+			} else if (removeDeletedUsers && !ldapUser) {
```

Comment:
ldapUser being undefined doesn't mean the user was deleted. If the ldap search returns multiple entries, `findLDAPUser` will also return undefined. There are some situations where LDAP may return multiple entries for the same user - we don't support this situation for login, but we should probably only delete users when they really weren't found.